lib/util/source-code-util.js

```diff
@@ -84,14 +84,14 @@ function getSourceCodeOfFiles(patterns, options, cb) {
     debug("constructed options:", opts);
     patterns = globUtil.resolveFileGlobPatterns(patterns, opts);
 
-    const filenames = globUtil.listFilesToProcess(patterns, opts).reduce(function(files, fileInfo) {
-        return !fileInfo.ignored ? files.concat(fileInfo.filename) : files;
-    }, []);
+    const filenames = globUtil.listFilesToProcess(patterns, opts)
+        .filter(fileInfo => !fileInfo.ignored)
+        .reduce((files, fileInfo) => files.concat(fileInfo.filename), []);
```

Comment:
I modified the logic here to prevent `no-confusing-arrow` from reporting an error.